cpp/src/arrow/gpu/cuda_context.cc

```diff
@@ -177,6 +205,25 @@ class CudaDeviceManager::CudaDeviceManagerImpl {
     return Status::OK();
   }
 
+  Status GetSharedContext(int device_number, CUcontext ctx,
+                          std::shared_ptr<CudaContext>* out) {
+    auto it = contexts_.find(device_number);
+    if (it == contexts_.end()) {
+      std::shared_ptr<CudaContext> new_context;
+      RETURN_NOT_OK(CreateSharedContext(device_number, ctx, &new_context));
+      contexts_[device_number] = *out = new_context;
+    } else {
+      if ((CUcontext)it->second.get()->handle() != ctx) {
```

Comment:
Use C++-style cast

Reply:
Now using `it->second.get()->handle() != static_cast<void*>(ctx)`.